File: apps/web/src/types/assignment.ts
export type AssignmentCourse = {
  id: number
  code: string
  name: string
  color: string
}

export type Assignment = {
  id: number
  course_id: number
  title: string
  due_date: string
  completed: boolean
  created_at: string
  course: AssignmentCourse
}

export type AssignmentInput = {
  course_id: number
  title: string
  due_date: string
  completed?: boolean
}

export type AssignmentUpdateInput = Partial<AssignmentInput>

File: apps/api/app/models/assignment.py
from datetime import date, datetime
from typing import TYPE_CHECKING

from sqlalchemy import Boolean, Date, DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base

if TYPE_CHECKING:
    from app.models.course import Course


class Assignment(Base):
    __tablename__ = "assignments"

    id: Mapped[int] = mapped_column(primary_key=True)
    course_id: Mapped[int] = mapped_column(
        ForeignKey("courses.id", ondelete="CASCADE"),
        index=True,
    )
    title: Mapped[str] = mapped_column(String(160))
    due_date: Mapped[date] = mapped_column(Date, index=True)
    completed: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    course: Mapped["Course"] = relationship(back_populates="assignments")
